### PR TITLE
#160604341 add user ability to sign up

### PR DESCRIPTION
**Pull request by jillo-abdullahi**
**What does this PR do?**
Consume api endpoint for user signup

**Tasks to be completed**
User ability to signup

**What are the associated Pivotal Tracker stories?**
#160604341

### CODE REVIEW

File: ui/js/signup.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        method: 'POST',
+        headers: {
+            'Accept': 'application/json, text/plain, */*',
+            'Content-type': 'application/json'
+        },
+        body:JSON.stringify(data)
+    })
+    .then((res) => res.json())
+    .then((data) => {
+        if (data.message == "user registered successfully"){
```
Comment (Dhul-Husni) [inline ui/js/signup.js]: use status code instead

File: ui/js/signup.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    let password = document.querySelector("#psw").value
+    let confirmPassword = document.querySelector("#cpsw").value
+
+    data = {
+            "username": username,
+            "email": email,
+            "password": password,
+            "confirm-password": confirmPassword
+        }
+    fetch('http://127.0.0.1:5000/stackoverflowlite/api/v1/auth/signup', {
```
Comment (Dhul-Husni) [inline ui/js/signup.js]: it would be better if we wrapped this into its own function and call it from there
ie 
signUpUser()

**Review by Dhul-Husni — CHANGES_REQUESTED** [review]
nice work, abit to look at

File: ui/js/signup.js (hunk 2 above)
Comment (jillo-abdullahi) [reply ui/js/signup.js]: This has been implemented. Please check latest commit. Gracias.

File: ui/js/signup.js (hunk 1 above)
Comment (jillo-abdullahi) [reply ui/js/signup.js]: Switched to using status codes instead. Please check last commit. Thanks.

**Review by Dhul-Husni — APPROVED** [review]
lgtm

**Pull request merged**